lib/srv/reexec.go

```diff
@@ -866,16 +866,24 @@ func getConnFile(conn net.Conn) (*os.File, error) {
 	}
 }
 
-// runCheckHomeDir check's if the active user's $HOME dir exists.
+// runCheckHomeDir check's if the active user's $HOME dir exists and is accessible.
 func runCheckHomeDir() (errw io.Writer, code int, err error) {
-	home, err := os.UserHomeDir()
+	currentUser, err := user.Current()
 	if err != nil {
-		return io.Discard, teleport.HomeDirNotFound, nil
+		return io.Discard, teleport.HomeDirNotAccessible, nil
```

Comment:
This looks incorrect per this comment:
```
// HomeDirNotAccessible is returned when a the "teleport checkhomedir" command has
// found the user's home directory, but the user does NOT have permissions to
// access it.
```

Reply:
Yeah I agree this is a confusing response. I think a generic `RemoteCommandFailure` might make more sense here since `user.Current()` returning an error would be fairly unexpected and I'm not sure adding a new error code would be worth it